.github/workflows/publish-firefox-development.yml

```diff
@@ -0,0 +1,88 @@
+# For the Firefox development addon, people install it manually,
+# and updates are distributed via the JSON file created in this
+# action which is stored on GitHub pages.
+
+name: publish-firefox-development
+on:
+  release:
+    types: [prereleased]
+jobs:
+  build-signed-xpi-asset:
+    needs: ensure-zip
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: cardinalby/export-env-action@v1
+        with:
+          envFile: "./.github/workflows/constants.env"
+          expand: true
+
+      - uses: robinraju/release-downloader@v1.7
+        with:
+          release: ${{ github.event.release.id }}
+          fileName: "*"
+
+      - name: Sign Firefox xpi for offline distribution
+        id: ffSignXpi
+        continue-on-error: true
+        uses: cardinalby/webext-buildtools-firefox-sign-xpi-action@v1
+        with:
+          timeoutMs: 1200000
+          extensionId: ${{ secrets.FF_OFFLINE_EXT_ID }}
+          zipFilePath: builds/yomitan-firefox-dev.zip
+          xpiFilePath: builds/yomitan-firefox-dev.xpi
+          jwtIssuer: ${{ secrets.FF_JWT_ISSUER }}
+          jwtSecret: ${{ secrets.FF_JWT_SECRET }}
+
+      - name: Abort on sign error
+        if: |
+          steps.ffSignXpi.outcome == 'failure' &&
+          steps.ffSignXpi.outputs.sameVersionAlreadyUploadedError != 'true'
+        run: exit 1
+
+      - name: Upload offline xpi release asset
+        id: uploadReleaseAsset
+        if: steps.ffSignXpi.outcome == 'success'
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ github.event.release.upload_url }}
+          asset_path: builds/yomitan-firefox.xpi
+          asset_name: yomitan-firefox.xpi
+          asset_content_type: application/x-xpinstall
+
+      # update update.json so that all people who have the dev version installed get the new update
+
+      - uses: actions/checkout@v3
+        with:
+          ref: metadata
+
+      - name: Recreate update.json
+        run: |
+          cat > update.json << EOF
+          {
+              "addons": {
+                  "themoeway+development@googlegroups.com": {
+                      "updates": [
+                          {
+                              "version": "${{ github.event.release.name }}",
+                              "update_link": "${{ uploadReleaseAsset.browser_download_url }}"
+                          }
+                      ]
+                  }
+              }
+          }
+          EOF
+
+      - name: Commit files
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git commit -a -m "Update"
```

Comment:
I wonder if we could do better by pulling a list of changes from the commit history here instead and create a better commit message

Reply:
Hmm, you mean to leave like a changelog in the commit? We could pull it from the release description. But I'm not sure it's necessary since normally no one would look at the branch lol. Maybe just the verison number and/or a link to the release or something

Reply:
Gave the version number + link idea a shot, let me know what you think. https://github.com/themoeway/yomitan/pull/92/commits/765e229965838dfcb26f3afeef950aeba22a5d3d#diff-8d5cfe2441357d30ded1a905f29730cd09ae031e3c8e850327fea566ae2b4109R78

Reply:
Yeah this looks better than the generic "Update" message for sure